### PR TITLE
LPS-68988 User is not reindexed when adding it to a Organization using UserLocalServiceUtil.addOrganizationUser API

### PR DESCRIPTION
**Pull request by pei-jung**
Hey Jorge,

This is an update for https://issues.liferay.com/browse/LPS-68988. 

From @jorgediaz-lr 's comment in the ticket 

> A user reindex was added in [LPS-378](https://github.com/liferay/liferay-portal/commit/4391692ff18ebaa047634f9dd541ad76d100c077) for addOrganizationUsers but it was not applied to addOrganizationUser

I was curious why we only do reindex for `addOrganizationUsers(long organizationId, long[] userIds)` but not other overloaded methods, so I did some research and it looks like the code has been like this for a long time. I think there might have been some changes to the service builder and we forgot to take into account the other overloaded methods (since those are autogenerated in BaseImpl).

I extended @jorgediaz-lr's fix to group, role, team, and user group since they are also being tracked by UserIndexer.

/cc @drewbrokke

### CODE REVIEW

**Comment by pei-jung** [comment]
ci:stop

**Comment by pei-jung** [comment]
ci:retest

File: portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java [hunk 1]
```diff
@@ -763,6 +763,59 @@ public User addUser(
...
+	 * @param userGroupId the primary key of the user group
+	 * @param userIds the primary key of the user
+	 */
+	@Override
+	@SuppressWarnings("deprecation")
+	public void addUserGroupUser(long userGroupId, long userId)
+		throws PortalException {
+
+		if (PropsValues.USER_GROUPS_COPY_LAYOUTS_TO_USER_PERSONAL_SITE) {
+			userGroupLocalService.copyUserGroupLayouts(userGroupId, userId);
```
Comment (pei-jung) [inline portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java]: I noticed [userGroupLocalService.copyUserGroupLayouts](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/service/impl/UserGroupLocalServiceImpl.java#L225-L229) has been deprecated for a while now. Should we come back and take look at this later when we get a chance?

Comment (JorgeFerrer) [inline portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java]: Yup, good point.

We should start by doing some due diligence and asking if customers are still using it. Support might be a first place to ask. Do you want to start that conversation?

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by JorgeFerrer** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/44517. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by JorgeFerrer** [comment]
Good job fixing this @jorgediaz-lr 

Thanks for being proactive and fixing it elsewhere as well @pei-jung :)

File: portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java (hunk 1 above)
Comment (pei-jung) [inline portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java]: Sure. Who will be the best person to ask in support?

Comment (JorgeFerrer) [inline portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java]: Maybe you can reach out to Minchau for advice.

**Pull request closed without merging**